### PR TITLE
Add CI Workflow

### PR DESCRIPTION
**Pull request by Mr0grog**
This adds a continuous integration workflow that runs linters and tests on pull requests (and on pushes to `main` and `staging`) and reports the results as status checks. It should help make obvious that a PR is not safe before clicking the merge so we don’t have failed deploys or need things like #74 to quickly fix a bad merge.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Mr0grog) [inline .github/workflows/ci.yml:71]: This issue is worth noting: while the app will take its database info from the environment variables, the *tests* hard-code a user and database name, and never use a password at all in the first place.

This configuration matches the hardcoded database name and user for the server tests (but not the `server-period-1` and `server-period-2` tests, which hardcode different names, and which I’m not sure are really meant to be used…?). It also sets `POSTGRES_HOST_AUTH_METHOD` to `trust` so that a password is not needed, since the tests hardcode not using a password. All of this is less than ideal, but at least it mostly works.

File: .github/workflows/ci.yml
Comment (Mr0grog) [inline .github/workflows/ci.yml:98]: Using the `.env.example` file this way doesn’t feel especially great, but it works. It might be better to explicitly set all the environment variables in the `env:` YAML block just below this, but the test scripts are also coded to expect a `.env` file to always be present.

File: tests/server/mocha_wrapper.sh
Comment (Mr0grog) [inline tests/server/mocha_wrapper.sh:3]: I had to add an explicit username here, otherwise the user is arbitrarily determined based on the linux user you are logged into the current linux machine as, which is something that might change from CI run to CI run. This happens to work as well for me locally (I use Postgres.app, and it always includes a `postgres` user in the database), but **it could break for other people.**

If this is a concern, we need to do some work so that this script loads some database configuration from the environment rather than hardcoding it all here.

**Review by Mr0grog — COMMENTED** [review]
This *is* working now, although there are some things that are less than stellar here and which may or may not warrant more work before merging. I tried to note these concerns inline, but I’ll leave it up to reviewers to decide what should be done (or not done).

**Comment by Mr0grog** [comment]
Once this lands, you may want to comment `@dependabot rebase` on all the pending Dependabot PRs. That way they’ll pick up on this new CI workflow and run all the tests.

File: tests/server/mocha_wrapper.sh
Comment (ajhyndman) [reply tests/server/mocha_wrapper.sh:3]: This seems like a step forward to me.  I am happy to merge as is.

File: .github/workflows/ci.yml
Comment (ajhyndman) [inline .github/workflows/ci.yml:14]: Nitpick: It looks like only one of these jobs has a name.  We could probably drop this name for consistency?

**Review by ajhyndman — APPROVED** [review]
This looks great to me!  Thanks very much for taking the time to set it up, @Mr0grog.

**Comment by ajhyndman** [comment]
fixes #81

**Comment by Mr0grog** [comment]
Looks like the server tests may be a bit flaky. :(

Otherwise this should be good to go! I’ll go ahead and merge later this afternoon if I don’t hear any concerns from you about the flakiness.

**Comment by ajhyndman** [comment]
I'm happy to merge!  We may end up rewriting the "server tests" entirely, anyway.

**Pull request merged**
